src/test/scala/uk/gov/nationalarchives/LambdaTest.scala

```diff
@@ -35,361 +25,68 @@ class LambdaTest extends AnyFlatSpec with BeforeAndAfterEach {
-      |        "S": "UpstreamSystemReference"
-      |      }
-      |    }
-      |  ]
-      |}
-      |""".stripMargin
-  val dynamoGetResponse: String =
-    s"""{
-       |  "Responses": {
-       |    "test-table": [
-       |      {
-       |        "id": {
-       |          "S": "$assetId"
-       |        },
-       |        "name": {
-       |          "S": "Test Name"
-       |        },
-       |        "parentPath": {
-       |          "S": "$assetParentPath"
-       |        },
-       |        "type": {
-       |          "S": "Asset"
-       |        },
-       |        "batchId": {
-       |          "S": "$batchId"
-       |        },
-       |        "transferringBody": {
-       |          "S": "Test Transferring Body"
-       |        },
-       |        "transferCompleteDatetime": {
-       |          "S": "2023-08-01T00:00Z"
-       |        },
-       |        "upstreamSystem": {
-       |          "S": "Test Upstream System"
-       |        },
-       |        "digitalAssetSource": {
-       |          "S": "Test Digital Asset Source"
-       |        },
-       |        "digitalAssetSubtype": {
-       |          "S": "Test Digital Asset Subtype"
-       |        },
-       |        "originalFiles": {
-       |          "L": [ { "S" : "b6102810-53e3-43a2-9f69-fafe71d4aa40" } ]
-       |        },
-       |        "originalMetadataFiles": {
-       |          "L": [ { "S" : "c019df6a-fccd-4f81-86d8-085489fc71db" } ]
-       |        },
-       |        "id_Code": {
-       |          "S": "Code"
-       |        },
-       |        "id_UpstreamSystemReference": {
-       |          "S": "UpstreamSystemReference"
-       |        }
-       |      }
-       |    ]
-       |  }
-       |}
-       |""".stripMargin
-
-  case class TestLambda() extends Lambda {
-    val creds: StaticCredentialsProvider = StaticCredentialsProvider.create(AwsBasicCredentials.create("test", "test"))
-    private val asyncDynamoClient: DynamoDbAsyncClient = DynamoDbAsyncClient
-      .builder()
-      .endpointOverride(URI.create("http://localhost:9005"))
-      .region(Region.EU_WEST_2)
-      .credentialsProvider(creds)
-      .build()
-
-    private val asyncS3Client: S3AsyncClient = S3AsyncClient
-      .crtBuilder()
-      .endpointOverride(URI.create("http://localhost:9006"))
-      .region(Region.EU_WEST_2)
-      .credentialsProvider(creds)
-      .targetThroughputInGbps(20.0)
-      .minimumPartSizeInBytes(10 * 1024 * 1024)
-      .build()
-    override val dynamoClient: DADynamoDBClient[IO] = new DADynamoDBClient[IO](asyncDynamoClient)
-    override val s3Client: DAS3Client[IO] = DAS3Client[IO](asyncS3Client)
-    override def getXmlCreator: XMLCreator = XMLCreator(OffsetDateTime.parse("2023-09-01T00:00Z"))
-  }
-
   "handleRequest" should "return an error if the asset is not found in dynamo" in {
-    stubGetRequest(emptyDynamoGetResponse)
+    val testUtils = new ExternalServicesTestUtils(dynamoServer, s3Server)
```

Comment:
You could cut down a bit on the repetition a bit by having this at the top of the class. 
You could make it even shorter by doing something like
```scala
val testUtils = new ExternalServicesTestUtils(dynamoServer, s3Server)
import testUtils._
```

Then you can get rid of all the `testUtils.` 

There's a few places where you've got
```scala
val assetId = testUtils.assetId
val assetParentPath = testUtils.assetParentPath
```
You could remove those lines completely and use assetId directly from testUtils.

Reply:
Oh cool, I did not know about the import thing. 👍